src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -590,15 +590,13 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
             return false;
         }
 
-        internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, BindingDiagnosticBag diagnostics)
+        internal override void AfterAccessorBindingChecks()
         {
             // This check prevents redundant ManagedAddr diagnostics on the underlying pointer field of a fixed-size buffer
             if (!IsFixedSizeBuffer)
             {
-                Type.CheckAllConstraints(DeclaringCompilation, conversions, ErrorLocation, diagnostics);
+                Type.CheckAllConstraints(DeclaringCompilation, new TypeConversions(this.ContainingAssembly.CorLibrary), ErrorLocation, DeclaringCompilation.AfterAccessorBindingDiagnostics);
```

Comment:
>DeclaringCompilation.AfterAccessorBindingDiagnostics [](http://example.com/codeflow?start=135&length=52)

What scenario does this change affect?

Reply:
This change resulted from using `AfterAccessorBindingChecks` instead of `AfterAddingTypeMembersChecks` to avoid extra binding. We now calculate the diagnostics more late than we used to, and we use `AfterAccessorBindingDiagnostics` for that.

The use of `AfterAccessorBindingChecks` instead of `AfterAddingTypeMembersChecks` affects the number of extra bindings performed in `TestERR_ManagedAddr`.

Does that answer the question or there is something I'm missing?